### ISSUE TITLE
Issue with self signed certificate's gitlab

### ISSUE THREAD

**Issue by herisson-88** [Blarc/gitlab-template-lint-plugin#39]
![image](https://user-images.githubusercontent.com/35228052/196120563-a30316ce-7b4c-4967-b65f-4b84832ebac4.png)
![image](https://user-images.githubusercontent.com/35228052/196121380-e846fd9d-9dc3-4910-b350-be0c9e70446a.png)

Seems having issue with self signed certificate (even if it has been accepted in Intellij)

**Comment by jamesdehart** [comment]
I ran into this issue. You need to add the root ca of your company to the Intellij's JRE cacerts file.

From my mac:
```sh
keytool -import -trustcacerts -noprompt -keystore /Applications/IntelliJ\ IDEA.app/Contents/jbr/Contents/Home/lib/security/cacerts  -storepass changeit -alias decrypt -file Downloads/_COMPANY_CA_CERT.cer
```

For Windows/Linux see --> https://intellij-support.jetbrains.com/hc/en-us/articles/206544879-Selecting-the-JDK-version-the-IDE-will-run-under

**Comment by Blarc** [comment]
I am using `InsecureTrustManager` for making requests: https://github.com/Blarc/gitlab-template-lint-plugin/blob/main/src/main/kotlin/com/github/blarc/gitlab/template/lint/plugin/gitlab/http/InsecureTrustManager.kt.
If you think this is a problem and know how to fix it, I will gladly accept PR or any advice on this topic.

Thanks.

**Comment by dvdgsng** [comment]
When using Jetbrains Toolbox on Linux, the cacerts file is burried deep into the ~/.local directory. Here the command for IntelliJ IDEA 2022.2.3 (Build #IU-222.4345.14):

```
keytool -import -trustcacerts -noprompt -keystore ~/.local/share/JetBrains/Toolbox/apps/IDEA-U/ch-0/222.4345.14/jbr/lib/security/cacerts  -storepass changeit -alias decrypt -file my-selfsigned.crt
```

You prob. need to update the path & run this again after each IntelliJ IDEA update.

**Comment by Blarc** [comment]
I have added a setting for allowing self-signed certificate. You can download the new version of the plugin here: https://github.com/Blarc/gitlab-template-lint-plugin/suites/12518028263/artifacts/668106543 (both should work) and install it by following the [steps](https://github.com/Blarc/gitlab-template-lint-plugin#installation-from-zip).

Please report back it this solves the problem. :pray:

**Comment by Blarc** [comment]
I have released a new version [1.10.0](https://github.com/Blarc/gitlab-template-lint-plugin/releases/tag/v1.10.0) that contains a setting for allowing self-signed certificate. You can wait for JetBrains to approve the new version and publish it on marketplace or install it manually by following these [steps](https://github.com/Blarc/gitlab-template-lint-plugin#installation-from-zip).

Thanks for reporting the issue. :100:
